### PR TITLE
Upgrade to ember-cli 3.0.0

### PR DESCRIPTION
**Pull request by olleolleolle**
This PR uses ember-cli 3.0.0, and changes the tests to support that. 

The tests pass locally for me, so I made this change to see if they'd run on Travis, too. (Now they do.)

Fixes:

- use newer testing infrastructure
- use strict
- define the 2.18 scenario (for the ember try:each)
- changed the acceptance test to not use "find()", which was not available

---

I used Kelly Selden's  [ember-cli-update](https://www.npmjs.com/package/ember-cli-update) to do the upgrade.

### CODE REVIEW

**Comment by nanderss** [comment]
We should not force the addon to use 3.0, since that will limit the possibilities for using it.

**Comment by olleolleolle** [comment]
@nanderss I may be confused, but those are `devDependencies`, right?

My speculative understanding, as a Ruby metaphor: They're like the Gemfile, whereas the `dependencies` are the .gemspec? Is this a fair description?

In the Ember test scenarios, we swap out 3.0.0 for various values, and the "ember test:each" will build "package.json" variants for each scenario.

File: tests/acceptance/ember-i18n-fetch-translations-test.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
-test('visiting /ember-i18n-fetch-translations', function(assert) {
-  visit('/');
-
-  andThen(function() {
-    assert.equal(find('div[data-illumination]').text(), 'Blinkenlichts');
-  });
+    test('visiting /ember-i18n-fetch-translations', async function(assert) {
+        await visit('/');
+        assert.equal(this.element.querySelector('div[data-illumination]').textContent, 'Blinkenlichts');
```
Comment (mriska) [inline tests/acceptance/ember-i18n-fetch-translations-test.js:10]: Doesn't `find` exist anymore?
Did you try 
```javascript
import { visit, find } from '@ember/test-helpers';
```

**Comment by mriska** [comment]
I believe @olleolleolle is correct on the 3.0.0 issue.

**Comment by nanderss** [comment]
Ok, my bad.

File: tests/acceptance/ember-i18n-fetch-translations-test.js (hunk 1 above)
Comment (olleolleolle) [reply tests/acceptance/ember-i18n-fetch-translations-test.js:10]: I didn't - and could!

Comment (olleolleolle) [reply tests/acceptance/ember-i18n-fetch-translations-test.js:10]:
```
not ok 5 Chrome 64.0 - Acceptance | ember i18n fetch translations: visiting /ember-i18n-fetch-translations
    ---
        actual: >
            null
        stack: >
            TypeError: (0 , _testHelpers.find) is not a function
                at Object.<anonymous> (http://localhost:7357/assets/tests.js:11:48)
                at <anonymous>
        message: >
            Promise rejected during "visiting /ember-i18n-fetch-translations": (0 , _testHelpers.find) is not a function
        Log: |
    ...```

Comment (olleolleolle) [reply tests/acceptance/ember-i18n-fetch-translations-test.js:10]: Is this because I'm defining an _asynchronous_ test?

**Comment by olleolleolle** [comment]
So, I'm like "this is a green build, while master is not".

**Review by mriska — APPROVED** [review]

**Pull request merged**